eZ/Publish/Core/FieldType/RichText/InternalLinkValidator.php

```diff
@@ -28,6 +30,40 @@ public function __construct(ContentService $contentService, LocationService $loc
         $this->locationService = $locationService;
     }
 
+    /**
+     * Extracts and validate internal links.
+     *
+     * @param \DOMDocument $xml
+     * @return array
+     */
+    public function validateDocument(DOMDocument $xml)
```

Comment:
However please use naming consistent with `eZ\Publish\Core\FieldType\RichText\Validator::validate`. Your naming choice is better, but I prefer to have this aligned. On the other hand, as it's Core, we could change `validate` to `validateDocument`, but it's still risky, so +0.25 on the latter suggestion ;)

Reply:
This comment still stands:
> However please use naming consistent with eZ\Publish\Core\FieldType\RichText\Validator::validate. Your naming choice is better, but I prefer to have this aligned. On the other hand, as it's Core, we could change validate to validateDocument, but it's still risky, so +0.25 on the latter suggestion ;)

I think to avoid changes out of scope, we should stick to `validate` name for consistency.

Reply:
Are you sure? I would like to but I've chosen `validateDocument` instead of `validate` because such method already exists in this class: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/FieldType/RichText/InternalLinkValidator.php#L41 (added in https://github.com/ezsystems/ezpublish-kernel/commit/62f36b37bf4e921a139021a8363b9f53363fedbd) and it's used in https://github.com/ezsystems/ezpublish-kernel/pull/2154/files#diff-f61b9ca9ce91bcb42d65bea5d6bc98b5R58

Reply:
Ah, sorry, sometimes it's good to see whole file instead of diff. My bad.